File: php/features.php
<?php
declare(strict_types=1);

// MarketData SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class MarketDataFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new MarketDataBaseFeature();
            case "test":
                return new MarketDataTestFeature();
            default:
                return new MarketDataBaseFeature();
        }
    }
}

File: py/features.py
# MarketData SDK feature factory

from feature.base_feature import MarketDataBaseFeature
from feature.test_feature import MarketDataTestFeature


def _make_feature(name):
    features = {
        "base": lambda: MarketDataBaseFeature(),
        "test": lambda: MarketDataTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
